src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts

```diff
@@ -24,36 +29,101 @@ export class OsdListComponent implements OnInit {
   statusColor: TemplateRef<any>;
   @ViewChild('osdUsageTpl')
   osdUsageTpl: TemplateRef<any>;
+  @ViewChild('markOsdConfirmationTpl')
+  markOsdConfirmationTpl: TemplateRef<any>;
+  @ViewChild('criticalConfirmationTpl')
+  criticalConfirmationTpl: TemplateRef<any>;
   @ViewChild(TableComponent)
   tableComponent: TableComponent;
+  @ViewChild('reweightBodyTpl')
+  reweightBodyTpl: TemplateRef<any>;
+  @ViewChild('safeToDestroyBodyTpl')
+  safeToDestroyBodyTpl: TemplateRef<any>;
 
   permission: Permission;
   tableActions: CdTableAction[];
   bsModalRef: BsModalRef;
-  osds = [];
   columns: CdTableColumn[];
+
+  osds = [];
   selection = new CdTableSelection();
 
+  protected static collectStates(osd) {
+    return [osd['in'] ? 'in' : 'out', osd['up'] ? 'up' : 'down'];
+  }
+
   constructor(
     private authStorageService: AuthStorageService,
     private osdService: OsdService,
     private dimlessBinaryPipe: DimlessBinaryPipe,
     private modalService: BsModalService
   ) {
     this.permission = this.authStorageService.getPermissions().osd;
-    const scrubAction: CdTableAction = {
-      permission: 'update',
-      icon: 'fa-stethoscope',
-      click: () => this.scrubAction(false),
-      name: 'Scrub'
-    };
-    const deleteAction: CdTableAction = {
-      permission: 'update',
-      icon: 'fa-cog',
-      click: () => this.scrubAction(true),
-      name: 'Deep Scrub'
-    };
-    this.tableActions = [scrubAction, deleteAction];
+    this.tableActions = [
+      {
+        name: 'Scrub',
+        permission: 'update',
+        icon: 'fa-stethoscope',
+        click: () => this.scrubAction(false),
+        disable: () => !this.hasOsdSelected
+      },
+      {
+        name: 'Deep Scrub',
+        permission: 'update',
+        icon: 'fa-cog',
+        click: () => this.scrubAction(true),
+        disable: () => !this.hasOsdSelected
+      },
+      {
+        name: 'Reweight',
+        permission: 'update',
+        click: () => this.reweight(),
+        disable: () => !this.hasOsdSelected,
+        icon: 'fa-balance-scale'
+      },
+      {
+        name: 'Mark Out',
+        permission: 'update',
+        click: () => this.markOut(),
```

Comment:
How about removing those one line functions and replace their call with the one line?
```suggestion
        click: () => this.showConfirmationModal('out', this.osdService.markOut);,
```

Reply:
I'd prefer to keep them, as it's clear what they do and it might not instantly be so clear what the `showConfirmationModal` method does. I see those one line methods as kind of additional documentation if you will. A hint where I can see what's actually supposed to happen, without having to write a documentation about it, it's just descriptive code.

Reply:
It could be unclear as the use of `showConfirmationModal` is much clearer as it hits the nail on the head what it is doing ;) `markOut` does not provide the information that a confirmation modal will be opened.

Reply:
Let's put it this way, the call of `showConfirmationModal` needs some parameters one has to look up, but `markOut` can just be used as it is. It provides an easy interface and encourages reuse :)

Reply:
Honestly... I would be fine with both solutions. I think  @Devp00l's idea is giving a bit more information about what will happen, but I (personally) would not mind looking up what's ```markOut``` about.

Reply:
Let me also provide another example where I directly pick up your (and originally my) argument. The name of a method needs to fit it's purpose, but that alone doesn't mean it provides a nice interface. Let's use the `requests` library from Python as an example. The name `request` is a perfect match for what the method does, but it's first argument is the desired method. One could write `requests.request('GET', ...)` all the time, or just use the one that provides a nicer interface, namely the `get` method. And that's actually what's more frequently used (in our code), except for the case where one needs the flexibility of `request`. And that's an example where no additional argument is involved, which would also be abstracted away in our case.

Reply:
You will only look up the parameters if something is wrong with the implementation and it needs a change and looking up parameters in TS is real easy as it was developed with IDE lookup strategies in mind.

If I understand you correctly you prefer to go from the declaration of the action, to a method that just calls another method to go to the called method of the action item you want to amend instead of going straight from the action declaration to the used method? IMHO it's an unnecessary extra hop to get to the source and it makes you think there is more than one call in the method.

Reply:
Sorry just saw the other 2 comments.

@p-na But its an library? It needs to be super generic and needs to be backwards compatible. Here this is not the case as it's a component and not a service.

Reply:
Thanks for your suggestion but I think, we won't find a conclusion in this PR. If you want, you can open a PR which proposes the changes you've requested here to a broader audience. Personally I think, the effort put into this has already exceeded the benefit.

Reply:
The audience would be the same, but the changeset would be a lot smaller. I agree in doing that, but don't get angry with me for doing it, as this PR will spring up short after this gets in.

BTW please don't mark my conversations as resolved, if they are not changed in the code, as I have to open up all closed conversations to find the conversation I was looking for. I will resolve my conversation if they are resolved for me.